GAE/src/org/waterforpeople/mapping/dataexport/ExportImportUtils.java

```diff
@@ -86,27 +93,62 @@ public static String md5Digest(List<Row> rows, int lastColumnIndex)
         return StringUtil.toHexString(digest.digest());
     }
 
-    public static String formatDate(Date date) {
+    /*
+     * Format date into a date-time string.
+     */
+    public static String formatDateTime(Date date) {
         if (date != null) {
-            return DATE_FORMAT.get().format(date);
+            return DATE_TIME_FORMAT.get().format(date);
         } else {
             return "";
         }
     }
 
-    public static Date parseDate(String dateString) {
+    /*
+     * Parse date-time string
+     */
+    public static Date parseDateTime(String dateString) {
         if (dateString == null || dateString.equals("")) {
             return null;
         }
         try {
-            return new Date(Long.parseLong(dateString));
-        } catch (NumberFormatException nfe) {
-            try {
-                return DATE_FORMAT.get().parse(dateString);
-            } catch (ParseException pe) {
-                log.error("bad date format: " + dateString + "\n" + pe.getMessage(), pe);
-                return null;
-            }
+            return DATE_TIME_FORMAT.get().parse(dateString);
+        } catch (ParseException pe) {
+            log.error("bad date format: " + dateString + "\n" + pe.getMessage(), pe);
+        }
+        return null;
+    }
+
+    /*
+     * Format DateQuestion response in a human-readable way.
+     * This date will not contain time details (yyyy-MM-dd)
+     */
+    public static String formatDateResponse(String value) {
+        try {
+            Date date = new Date(Long.valueOf(value));
+            return DATE_RESPONSE_FORMAT.get().format(date);
+        } catch (NumberFormatException e) {
```

Comment:
Yes. exactly. and this was by design to handle the possibility of having date strings coming from the datastore and going out to the exported report as you can see in the comment of that commit.